components/CreateOrganizationForm.js

```diff
@@ -1,289 +1,387 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import PropTypes from 'prop-types';
-import { get } from 'lodash';
+import themeGet from '@styled-system/theme-get';
+import { Field, Form, Formik } from 'formik';
+import { trim } from 'lodash';
+import { withRouter } from 'next/router';
 import { defineMessages, FormattedMessage, injectIntl } from 'react-intl';
+import slugify from 'slugify';
+import styled from 'styled-components';
 
-import { defaultBackgroundImage } from '../lib/constants/collectives';
-
-import InputField from './InputField';
+import OnboardingProfileCard from './onboarding-modal/OnboardingProfileCard';
+import CollectivePickerAsync from './CollectivePickerAsync';
+import Container from './Container';
+import { Box, Flex } from './Grid';
+import MessageBox from './MessageBox';
 import StyledButton from './StyledButton';
 import StyledCheckbox from './StyledCheckbox';
+import StyledHr from './StyledHr';
+import StyledInput from './StyledInput';
+import StyledInputField from './StyledInputField';
+import StyledInputGroup from './StyledInputGroup';
 import StyledLink from './StyledLink';
+import StyledTextarea from './StyledTextarea';
+import { H1, H4, P } from './Text';
 
-class CreateOrganizationForm extends React.Component {
-  static propTypes = {
-    collective: PropTypes.object,
-    loading: PropTypes.bool,
-    onSubmit: PropTypes.func,
-    intl: PropTypes.object.isRequired,
-  };
-
-  constructor(props) {
-    super(props);
-    this.defineFields = this.defineFields.bind(this);
-    this.addLabels = this.addLabels.bind(this);
-    this.handleSubmit = this.handleSubmit.bind(this);
-    this.handleChange = this.handleChange.bind(this);
-    this.handleObjectChange = this.handleObjectChange.bind(this);
+const BackButton = styled(StyledButton)`
+  color: ${themeGet('colors.black.600')};
+  font-size: 14px;
+`;
 
-    const collective = { ...(props.collective || {}) };
-    collective.slug = collective.slug ? collective.slug.replace(/.*\//, '') : '';
+const messages = defineMessages({
+  nameLabel: { id: 'createOrg.form.nameLabel', defaultMessage: "What's the name of your organization?" },
+  slugLabel: { id: 'createOrg.form.slugLabel', defaultMessage: 'What URL would you like?' },
```

Comment:
Don't duplicate messages, for example instead of creating `createOrg.form.slugLabel` just reuse `createCollective.form.slugLabel` if they are exactly the same. You can search the `en.json` language file to check if the string you want to format already exists.

Reply:
Done